presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -84,15 +86,17 @@
     private final AtomicLong finishUserNanos = new AtomicLong();
 
     private final AtomicLong memoryReservation = new AtomicLong();
-    private final AtomicLong revocableMemoryReservation = new AtomicLong();
+    @GuardedBy("this")
```

Comment:
I re-rise the same issue. Having this class partially `ThreadSafe` is gigantic `wtf` for future programmers. It's either should be split or it should be `ThreadSafe` as a whole.